orchestrator/modules/tools/execution/exec_research.py

```
@@ -1,68 +1,171 @@
+
+    # PRD-160 S4: every NL query lands one audit row (best-effort).
+    try:
+        await service.write_nl_audit(
+            source_id=source_id,
+            user_id=user_id or None,
+            agent_id=agent,
+            nl_query=query,
+            result=result if isinstance(result, dict) else {},
+        )
+    except Exception:  # noqa: BLE001
+        pass
```

Ruff (0.15.15)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)